common/beerocks/bcl/include/bcl/network/socket.h

```diff
@@ -96,19 +151,22 @@ class Socket {
     bool m_accepted_socket  = false;
     bool m_external_handler = false;
     bool m_is_server        = false;
+    SOCKET getSocket() const { return m_socket; }
+    void setAcceptedSocket(bool value) { m_accepted_socket = value; }
+    void setUdsPath(const std::string &value) { m_uds_path = value; }
 
 private:
     static int m_ref;
 };
 
-class SocketClient : public Socket {
+class SocketClient : public DefaultSocket {
```

Comment:
This one and SocketServer will eventually have to be made mockable as well... which will bring us in the joys of multiple inheritance :-(

Anyway, better to focus on the easy cases first, so it's good to do it like this for now.